### PR TITLE
feat(plat): added support to Nuvoton npcm845x chip

### PR DESCRIPTION
**Pull request by rutigl**
feat(plat): added support to Nuvoton npcm845x chip

Initial version

Signed-off-by: Margarita Glushkin <rutigl@gmail.com>"

### CODE REVIEW

**Review by etienne-lms — COMMENTED** [review]
I had some pending comments I think should be addressed before one look into the implementation. Please consider them, including coding style issues that most of the time prevent one to focus on the implementation.

File: core/arch/arm/plat-nuvoton/npcm845x_utils.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+void print_version(void)
+{
+	TMSG(COLOR_MAGENTA);
+	TMSG(">================================================");
+	TMSG("OP-TEE OS Version %s", core_v_str);
+	TMSG("Nuvoton OP-TEE OS Version %d.%d.%d%s",
+			NPCMX845_OPTEE_VERSION_MAJOR,
+			NPCMX845_OPTEE_VERSION_MINOR,
+			NPCMX845_OPTEE_VERSION_PATCH,
+			NPCMX845_OPTEE_VERSION_BUILD);
```
Comment (dphadke-msft) [inline core/arch/arm/plat-nuvoton/npcm845x_utils.c:37]: Why is this defining custom version for Nuvoton, could use current optee version could be used `CFG_OPTEE_REVISION_{MAJOR,MINOR}`?

File: core/arch/arm/plat-nuvoton/npcm845x_trace.c
Comment (dphadke-msft) [inline core/arch/arm/plat-nuvoton/npcm845x_trace.c:1]: Why are these platform specific helpers required instead of `trace_printf()` and `trace_vprintf()`?

**Comment by rutigl** [comment]
Hi Dhanajay,
Custom version is not necessary, I removed it.
Platform specific helpers need for color output.
I didn't want to add color output to generic helpers, because others
don't need it.


2023-05-05 4:43 GMT+03:00, Dhananjay Phadke ***@***.***>:

**Comment by etienne-lms** [comment]
This P-R seems superseded by https://github.com/OP-TEE/optee_os/pull/5998.
If so, please close this P-R.

**Comment by rutigl** [comment]
This P-R seems superseded by [#5998].

**Comment by rutigl** [comment]
Done.

2023-05-09 12:59 GMT+03:00, Etienne Carriere ***@***.***>:

**Pull request closed without merging**